cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -109,72 +110,47 @@ public ProcessResult run(boolean capture, boolean isBackgroundProcess) {
-    boolean isBashScript = "sh".equals(fileExtension) || hasSheBang(this.executable);
-    if (isBashScript) {
-      String bash = "bash";
-      if (this.context.getSystemInfo().isWindows()) {
-        String findBashOnWindowsResult = findBashOnWindows();
-        if (findBashOnWindowsResult != null) {
-          bash = findBashOnWindowsResult;
-        }
-      }
-      this.arguments.add(0, bash);
-    }
-    this.processBuilder.command(this.arguments);
+
+    boolean isBashScript = checkAndHandlePossibleBashScript(executableName);
+
     if (this.context.debug().isEnabled()) {
       String message = createCommandMessage(" ...");
       this.context.debug(message);
     }
+
     try {
+
       if (capture) {
         this.processBuilder.redirectOutput(Redirect.PIPE).redirectError(Redirect.PIPE);
+      } else if (isBackgroundProcess) {
+        modifyArgumentsOnBackgroundProcess(isBashScript);
       }
+
+      this.processBuilder.command(this.arguments);
+
+      Process process = this.processBuilder.start();
+
       List<String> out = null;
       List<String> err = null;
-      Process process = this.processBuilder.start();
       if (capture) {
         out = new ArrayList<>();
         err = new ArrayList<>();
-        try (BufferedReader outReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
-            BufferedReader errReader = new BufferedReader(new InputStreamReader(process.getErrorStream()))) {
-          String outLine = "";
-          String errLine = "";
-          while ((outLine != null) || (errLine != null)) {
-            if (outLine != null) {
-              outLine = outReader.readLine();
-              if (outLine != null) {
-                out.add(outLine);
-              }
-            }
-            if (errLine != null) {
-              errLine = errReader.readLine();
-              if (errLine != null) {
-                err.add(errLine);
-              }
-            }
-          }
-        }
+        handleCapture(process, out, err);
       }
-      int exitCode = process.waitFor();
-      ProcessResult result = new ProcessResultImpl(exitCode, out, err);
-      if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
-        String message = createCommandMessage(" failed with exit code " + exitCode + "!");
-        if (this.errorHandling == ProcessErrorHandling.THROW) {
-          throw new CliException(message, exitCode);
-        }
-        IdeSubLogger level;
-        if (this.errorHandling == ProcessErrorHandling.ERROR) {
-          level = this.context.error();
-        } else if (this.errorHandling == ProcessErrorHandling.WARNING) {
-          level = this.context.warning();
-        } else {
-          level = this.context.error();
-          level.log("Internal error: Undefined error handling {}", this.errorHandling);
-        }
-        level.log(message);
+
+      int exitCode;
+      if (isBackgroundProcess) {
```

Comment:
This point probably is disputable

Reply:
We need to return `ProcessResult` and it has to have some exit code.
Surely we could also return different magic number for this or change exitCode to `Integer` and have it `null`.
But IMHO this is a reasonable behavior.
In bash the behavior is the same and the script shall continue successfully.
AFAIK if the command itself could not be started (e.g. because the command does not exist) then in bash you get an error but once it could be launched in background it is considered as success.
We can even remove the TODO here.

Reply:
I will resolve this conversation, but leave it open for today so you can remove the TODO.

Reply:
Done